controllers/serviceaccount_controller.go

```diff
@@ -224,9 +172,13 @@ func (r ServiceAccountReconciler) Reconcile(ctx goctx.Context, req reconcile.Req
 		}
 	}()
 
-	// This type of controller doesn't care about delete events.
-	if !vsphereCluster.DeletionTimestamp.IsZero() {
-		return r.ReconcileDelete(clusterContext)
+	pSvcAccounts, _ := getProviderServiceAccounts(clusterContext)
+	if len(pSvcAccounts) > 0 {
+		for _, acc := range pSvcAccounts {
+			if !acc.DeletionTimestamp.IsZero() {
+				return r.ReconcileDelete(clusterContext)
+			}
+		}
```

Comment:
What's the reason for the logic change? What is responsible for setting the DeletionTimestamp on the `ProviderServiceAccount` object? Usually that is performed by the ownerRef chain, but I don't think the ownerRef chain is used to setup the link between a pSvcAccount and a VSphereCluster object.

Plus, the Deletion logic looks for all the providerServiceAccounts associated to a VSphereCluster and removes them from the ConfigMap. Am not sure, if the DeletionTimestamp is set on the providerSvcAccount object at all. Can you verify this behavior? 
cc: @sidharthsurana can you confirm this logic?

Reply:
This code block seems not useful, since the method `getProviderServiceAccounts` already filters out the PSA that have deletionTimestamp set. Thus, you are guaranteed that you will never go inside the if block and execute the `ReconcileDelete` to begin with. I am not sure what might have prompted you to add this code here, but seems like safe to remove.

see: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/6535f8d5e22907cc27bd888f4a75ec2f0e0e4f25/controllers/serviceaccount_controller.go#L461-L481